python/SAGEMAKER_NOTEBOOK_KMS_CONFIGURED/SAGEMAKER_NOTEBOOK_KMS_CONFIGURED_test.py

```diff
@@ -53,9 +53,9 @@ class ComplianceTest(unittest.TestCase):
     no_notebook_instances_list = {"NotebookInstances": []}
     notebook_instances_list_mixed = [{'NotebookInstances': [{'NotebookInstanceName': 'trial12'}, {'NotebookInstanceName': 'trial123'}, {'NotebookInstanceName': 'trial1234'}]}]
 
-    described_notebook_instances = [{'NotebookInstanceName': 'trial12', 'KmsKeyId': 'arn:aws:kms:us-east-1:123456789012:key/7af97db7-f6a3-4d0a-87b9-a2737b54856d'}, {'NotebookInstanceName': 'trial123', 'KmsKeyId': 'arn:aws:kms:us-east-1:123456789012:key/7af97db7-f6a3-4d0a-87b9-a2737b54856d'}]
+    described_notebook_instances = [{'NotebookInstanceName': 'trial12', 'KmsKeyId': '7af97db7-f6a3-4d0a-87b9-a2737b54856d'}, {'NotebookInstanceName': 'trial123', 'KmsKeyId': '7af97db7-f6a3-4d0a-87b9-a2737b54856d'}]
```

Comment:
Good catch!

Reply:
I just run the test on an sagemaker notebook. It is actually the KMS key ARN which is given by sagemaker_client.describe_notebook_instance(), not the KMS key ID. There is a discrepancy between the actual value, the key and the documentation. But bottom line, the rule code should work.
Where did you run your test? I am currently running my test in ap-southeast-1.

Reply:
Hmm..that's very strange. I ran my test in us-east-1 and got KMS key ID and haven't tried it in other regions.

Reply:
I just run my test in us-east-1. I get a KmsKeyArn as well. Can you please tell the step to reproduce your test?

I am running a python3.7 lambda function

```
import json
import boto3

def lambda_handler(event, context):
    sagemaker_client = boto3.client('sagemaker')
    notebook_instance_description = sagemaker_client.describe_notebook_instance(NotebookInstanceName="test-for-rule")
    return {
        'statusCode': 200,
        'body': str(notebook_instance_description)
    }
```

Reply:
This PR doesn't seem like it's necessary.

I had created the SageMaker instance through terraform and used the KMS key ID when creating the instance the [CreateNotebookInstance API](https://docs.aws.amazon.com/sagemaker/latest/dg/API_CreateNotebookInstance.html#SageMaker-CreateNotebookInstance-request-KmsKeyId) accepted just the ID. Because I gave the key ID, that's what I saw I the response. I just created the instance again with the KMS ARN and the rule worked as expected.

Never went down this rabbit hole but do you know where I can see the repo that handles the REST APIs for SageMaker or is that entirely internal to AWS?

Reply:
Ok cool.

As far as I know, we didn't opensource the APIs handlers as such. After all, "APIs are forever" :)
You can see the read-only standard definition file in our SDKs though. For example, boto: https://github.com/boto/botocore/blob/develop/botocore/data/sagemaker/2017-07-24/service-2.json

Reply:
Good to know. Thanks for all your quick responses!